File: CrouskieBackOffice/src/crouskiebackoffice/model/Collection.java
package crouskiebackoffice.model;

/**
 * Classe modélisant une collection de vêtements.
 */
public class Collection {

    /**
     * L'ID de la collection dans la base de données. Valeur par défaut: -1
     */
    private int id = -1;
    /**
     * Le nom de la collection.
     */
    private String name;
    /**
     * Le chemin vers l'image de la collection.
     */
    private String pathPicture;

    /**
     * Crée une nouvelle instance de Collection en copiant les valeurs de
     * l'objet other.
     *
     * @param other l'objet à copier
     */
    public Collection(Collection other) {
        if (other != null) {
            this.id = other.id;
            this.name = other.name;
            this.pathPicture = other.pathPicture;
        }
    }

    /**
     * Crée une nouvelle instance de Collection avec l'ID et le nom spécifiés.
     *
     * @param id l'ID de la collection
     * @param name le nom de la collection
     */
    public Collection(int id, String name) {
        this.id = id;
        this.name = name;
    }

    /**
     * Constructeur.
     *
     * @param id l'ID de la collection
     * @param name le nom de la collection
     * @param pathPicture le chemin vers l'image de la collection
     */
    public Collection(int id, String name, String pathPicture) {
        this.id = id;
        this.name = name;
        this.pathPicture = pathPicture;
    }

    /**
     * Retourne le chemin vers l'image de la collection.
     *
     * @return Le chemin vers l'image de la collection.
     */
    public String getPathPicture() {
        return pathPicture;
    }

    /**
     * Retourne l'ID de la collection.
     *
     * @return L'ID de la collection.
     */
    public int getId() {
        return id;
    }

    /**
     * Modifie l'ID de la collection.
     *
     * @param id Le nouvel ID de la collection.
     */
    public void setId(int id) {
        this.id = id;
    }

    /**
     * Retourne le nom de la collection.
     *
     * @return Le nom de la collection.
     */
    public String getName() {
        return name;
    }

    /**
     * Modifie le nom de la collection.
     *
     * @param name Le nouveau nom de la collection.
     */
    public void setName(String name) {
        this.name = name;
    }

    /**
     * Modifie le chemin vers l'image de la collection.
     *
     * @param pathPicture Le nouveau chemin vers l'image de la collection.
     */
    public void setPathPicture(String pathPicture) {
        this.pathPicture = pathPicture;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: entities/Collection.php
<?php
/**
 * Cette classe représente un produit vendu par crouskie
 */
class Collection
{
    // ----------- Properties -------------

    private string $_name;
    private int $_collection_id;
    private string $_path_picture;

    /**
     * Constructeur de la classe Collection
     * @param string $name Le nom de la collection
     * @param int $collection_id L'identifiant de la collection
     * @param string $path_picture L'url de l'image associée à la collection
     */
    function __construct(string $name, int $collection_id, string $path_picture)
    {
        $this->_name = $name;
        $this->_collection_id = $collection_id;
        $this->_path_picture = $path_picture;
    }

    // ----------- Getters -------------
    
    /**
     * Récupère le nom de la collection
     * @return string Le nom de la collection
     */
    function getName() : string {
        return $this->_name;
    }

    /**
     * Récupère l'identifiant de la collection
     * @return int L'identifiant de la collection
     */
    function getCollectionId() : int {
        return $this->_collection_id;
    }

    /**
     * Récupère l'url de l'image associée à la collection
     * @return string L'url de l'image associé à la collection
     */
    function getPathPicture() : string
    {
        return $this->_path_picture;
    }
}
